ipv8/test/base.py

```diff
@@ -49,10 +51,14 @@ def setUp(self):
         self.__lockup_timestamp__ = time.time()
 
     def tearDown(self):
-        super(TestBase, self).tearDown()
-        for node in self.nodes:
-            node.unload()
-        internet.clear()
+        try:
+            super(TestBase, self).tearDown()
+            for node in self.nodes:
+                node.unload()
+            internet.clear()
+        finally:
+            shutil.rmtree("temp", ignore_errors=True)
```

Comment:
You are not removing the directory created in the method `temporary_directory`.

Reply:
`temporary_directory` creates a temporary directory in the `temp` folder (see line 139), this entire folder is `rmtree`'d.